test/integration-test/src/utils.rs

```diff
@@ -10,23 +11,26 @@ use aya::netlink_set_link_up;
 use libc::if_nametoindex;
 use netns_rs::{get_from_current_thread, NetNs};
 
-pub struct Netns {
-    pub name: String,
-    pub old_ns: NetNs,
-    pub ns: Option<NetNs>,
+pub struct NetNsGuard {
+    name: String,
+    old_ns: NetNs,
+    ns: Option<NetNs>,
 }
 
-impl Netns {
+impl NetNsGuard {
     pub fn new() -> Self {
         static COUNTER: AtomicU64 = AtomicU64::new(0);
         let pid = std::process::id();
         let name = format!("aya-test-{pid}-{}", COUNTER.fetch_add(1, Ordering::Relaxed));
+        let old_ns = get_from_current_thread().expect("Failed to get current netns");
```

Comment:
sorry to harp on this more: make this the first thing in the function?the preceding two lines are infallible.